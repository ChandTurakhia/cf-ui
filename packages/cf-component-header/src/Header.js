import React from 'react';
import PropTypes from 'prop-types';
import { createComponent } from 'cf-style-container';

const Header = createComponent(
  ({ theme, sticky }) => ({
    backgroundImage: theme.gradient.dawnSunrise,
    backgroundSize: '100% 3px',
    backgroundRepeat: 'no-repeat',
    backgroundColor: theme.color.moonshine,
    borderBottom: `1px solid ${theme.color.smoke}`,
    display: 'flex',
    justifyContent: 'space-between',
    color: theme.color.charcoal,
    height: 63,
    paddingTop: 3,
    top: 0,
    left: 0,
    position: sticky ? 'fixed' : 'static',
    width: '100%',
    borderBottom: `1px solid ${theme.color.dust}`,
    zIndex: 1200
  }),
  'header'
);

Header.propTypes = {
  children: PropTypes.node.isRequired,
  sticky: PropTypes.bool
};

export default Header;
