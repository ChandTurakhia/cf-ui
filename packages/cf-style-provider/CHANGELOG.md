# Change Log

All notable changes to this project will be documented in this file.
See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.

<a name="2.2.3"></a>
## [2.2.3](https://github.com/koddsson/cf-ui/compare/cf-style-provider@2.2.2...cf-style-provider@2.2.3) (2017-07-31)




<a name="2.2.2"></a>
## [2.2.2](https://github.com/cloudflare/cf-ui/compare/cf-style-provider@2.2.1...cf-style-provider@2.2.2) (2017-07-19)




<a name="2.2.0"></a>
# [2.2.0](https://github.com/sejoker/cf-ui/compare/cf-style-provider@2.1.2...cf-style-provider@2.2.0) (2017-07-12)


### Features

* **cf-style-provider:** allow passing selectPrefix in cf-style-provider ([3bc1ba9](https://github.com/sejoker/cf-ui/commit/3bc1ba9))




<a name="2.1.4"></a>
## [2.1.4](https://github.com/koddsson/cf-ui/compare/cf-style-provider@2.1.3...cf-style-provider@2.1.4) (2017-07-10)




<a name="2.1.3"></a>
## [2.1.3](https://github.com/koddsson/cf-ui/compare/cf-style-provider@2.1.2...cf-style-provider@2.1.3) (2017-07-10)




<a name="2.1.2"></a>
## [2.1.2](https://github.com/cloudflare/cf-ui/compare/cf-style-provider@2.1.0...cf-style-provider@2.1.2) (2017-07-06)




<a name="2.1.1"></a>
## [2.1.1](https://github.com/koddsson/cf-ui/compare/cf-style-provider@2.1.0...cf-style-provider@2.1.1) (2017-07-06)




<a name="2.1.0"></a>
# [2.1.0](https://github.com/cloudflare/cf-ui/compare/cf-style-provider@2.0.2...cf-style-provider@2.1.0) (2017-07-05)


### Features

* **cf-style-provider:** Print warnings in the dev mode if custom media queries are used ([b8a67c6](https://github.com/cloudflare/cf-ui/commit/b8a67c6))
* **cf-style-provider:** Remove monolithic mode from the dev configuration ([427bf80](https://github.com/cloudflare/cf-ui/commit/427bf80))
* **cf-style-provider:** Sort media queries, add better README ([d89f7fe](https://github.com/cloudflare/cf-ui/commit/d89f7fe))
* **cf-style-provider:** Update snapshots - they use atomic classNames now ([1a42284](https://github.com/cloudflare/cf-ui/commit/1a42284))
