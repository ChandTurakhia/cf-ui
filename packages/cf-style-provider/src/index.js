import createRenderer from './createRenderer';
import felaSnapshot from './felaSnapshot';
import felaTestContext from './felaTestContext';
import StyleProvider from './StyleProvider';

export { createRenderer, felaSnapshot, felaTestContext, StyleProvider };
