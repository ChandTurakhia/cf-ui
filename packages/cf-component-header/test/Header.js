import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../src/Header';
import { felaSnapshot } from 'cf-style-provider';

test('should render', () => {
  const snapshot = felaSnapshot(<Header>Hello World</Header>);
  expect(snapshot.component).toMatchSnapshot();
  expect(snapshot.styles).toMatchSnapshot();
});

test('should render sticky', () => {
  const snapshot = felaSnapshot(<Header sticky>Hello World</Header>);
  expect(snapshot.component).toMatchSnapshot();
  expect(snapshot.styles).toMatchSnapshot();
});
