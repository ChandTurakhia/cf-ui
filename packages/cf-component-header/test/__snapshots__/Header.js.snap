// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<header
  className="a b c d e f g h i j k l m n o"
>
  Hello World
</header>
`;

exports[`should render 2`] = `
"
.a {
  background-image: -webkit-linear-gradient(left, #F16975, #F69259);
  background-image: -moz-linear-gradient(left, #F16975, #F69259);
  background-image: linear-gradient(left, #F16975, #F69259)
}

.b {
  background-size: 100% 3px
}

.c {
  background-repeat: no-repeat
}

.d {
  background-color: #fff
}

.e {
  border-bottom: 1px solid #ebebeb
}

.f {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex
}

.g {
  justify-content: space-between;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between
}

.h {
  color: #333333
}

.i {
  height: 63px
}

.j {
  padding-top: 3px
}

.k {
  top: 0px
}

.l {
  left: 0px
}

.m {
  position: static
}

.n {
  width: 100%
}

.o {
  z-index: 1200
}
"
`;

exports[`should render sticky 1`] = `
<header
  className="a b c d e f g h i j k l m n o"
>
  Hello World
</header>
`;

exports[`should render sticky 2`] = `
"
.a {
  background-image: -webkit-linear-gradient(left, #F16975, #F69259);
  background-image: -moz-linear-gradient(left, #F16975, #F69259);
  background-image: linear-gradient(left, #F16975, #F69259)
}

.b {
  background-size: 100% 3px
}

.c {
  background-repeat: no-repeat
}

.d {
  background-color: #fff
}

.e {
  border-bottom: 1px solid #ebebeb
}

.f {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex
}

.g {
  justify-content: space-between;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between
}

.h {
  color: #333333
}

.i {
  height: 63px
}

.j {
  padding-top: 3px
}

.k {
  top: 0px
}

.l {
  left: 0px
}

.m {
  position: fixed
}

.n {
  width: 100%
}

.o {
  z-index: 1200
}
"
`;
